File: Codes from the AI-assistants/[Experimental] fannkuch-redux/Basic Prompt - TBR Mode/FannkuchRedux-copilot-basic-tbr.py
import itertools
import sys

def fannkuch_redux(n):
    max_flips = 0
    checksum = 0
    # permutations yields tuples of 0..n-1 in lexicographic order
    for idx, perm in enumerate(itertools.permutations(range(n))):
        # copy to list for mutable operations
        p = list(perm)
        # working copy for flipping so original perm isn't needed further
        q = p[:] 
        flips = 0
        while q[0] != 0:
            k = q[0] + 1
            # reverse prefix of length k in-place
            i, j = 0, k - 1
            while i < j:
                q[i], q[j] = q[j], q[i]
                i += 1
                j -= 1
            flips += 1
        max_flips = max(max_flips, flips)
        checksum += flips if (idx % 2 == 0) else -flips
    return checksum, max_flips

if __name__ == "__main__":
    n = int(sys.argv[1]) if len(sys.argv) > 1 else 7
    checksum, max_flips = fannkuch_redux(n)
    print(f"{checksum}\nPfannkuchen({n}) = {max_flips}")

File: Codes from the AI-assistants/[Experimental] fannkuch-redux/Basic Prompt - TBR Mode/FannkuchRedux-copilot-basic-tbr.js
// Node.js implementation with in-place next permutation (lexicographic)
function nextPermutation(arr) {
  // find pivot
  let i = arr.length - 2;
  while (i >= 0 && arr[i] >= arr[i + 1]) i--;
  if (i < 0) return false;
  let j = arr.length - 1;
  while (arr[j] <= arr[i]) j--;
  [arr[i], arr[j]] = [arr[j], arr[i]];
  // reverse suffix
  let a = i + 1, b = arr.length - 1;
  while (a < b) {
    [arr[a], arr[b]] = [arr[b], arr[a]];
    a++; b--;
  }
  return true;
}

function fannkuchRedux(n) {
  const p = Array.from({length: n}, (_, i) => i);
  let maxFlips = 0;
  let checksum = 0;
  let idx = 0;
  while (true) {
    // copy to working array for flips
    const q = p.slice();
    let flips = 0;
    while (q[0] !== 0) {
      const k = q[0] + 1;
      // reverse prefix k
      for (let i = 0, j = k - 1; i < j; i++, j--) {
        const t = q[i]; q[i] = q[j]; q[j] = t;
      }
      flips++;
    }
    if (flips > maxFlips) maxFlips = flips;
    checksum += (idx % 2 === 0) ? flips : -flips;
    idx++;
    if (!nextPermutation(p)) break;
  }
  return { checksum, maxFlips };
}

// Example run
const n = parseInt(process.argv[2] || "7", 10);
const result = fannkuchRedux(n);
console.log(result.checksum);
console.log(`Pfannkuchen(${n}) = ${result.maxFlips}`);